service/src/test/java/uk/gov/hmcts/reform/fpl/service/OrderValidationServiceTest.java

```diff
@@ -0,0 +1,106 @@
+import java.util.Map;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.OrderStatus.DRAFT;
+import static uk.gov.hmcts.reform.fpl.enums.OrderStatus.SEALED;
+import static uk.gov.hmcts.reform.fpl.service.HearingBookingService.HEARING_DETAILS_KEY;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBookings;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {OrderValidationService.class, JacksonAutoConfiguration.class})
+public class OrderValidationServiceTest {
+
+    @Autowired
+    private OrderValidationService validationService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    void shouldReturnErrorsWhenNoHearingDetailsExistsForSealedOrder() {
+        CaseDetails caseDetails = buildCaseDetails(SEALED);
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<String> returnedErrors = validationService.validate(caseData);
+
+        assertThat(returnedErrors)
+            .containsOnly("This standard directions order does not have a hearing associated with it. "
+                + "Please enter a hearing date and resubmit the SDO");
+    }
+
+    @Test
+    void shouldNotReturnErrorsWhenHearingDetailsExistsForSealedOrder() {
+        CaseDetails caseDetails = buildCaseDetailsWithHearingDetails(SEALED);
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<String> returnedErrors = validationService.validate(caseData);
+
+        assertThat(returnedErrors).isEmpty();
+    }
+
+    @Test
+    void shouldNotReturnErrorsWhenNoHearingDetailsExistsForSealedOrder() {
+        CaseDetails caseDetails = buildCaseDetails(DRAFT);
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<String> returnedErrors = validationService.validate(caseData);
+
+        assertThat(returnedErrors).isEmpty();
+    }
+
+    @Test
+    void shouldNotReturnErrorsWhenHearingDetailsExistsForDraftOrder() {
+        CaseDetails caseDetails = buildCaseDetailsWithHearingDetails(DRAFT);
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<String> returnedErrors = validationService.validate(caseData);
+
+        assertThat(returnedErrors).isEmpty();
+    }
+
+    private Order buildOrderWithStatus(final OrderStatus orderStatus) {
+        return Order.builder()
+            .orderStatus(orderStatus)
+            .build();
+    }
+
+    private CaseDetails buildCaseDetails(final OrderStatus orderStatus) {
+        return CaseDetails.builder()
+            .data(ImmutableMap.<String, Object>builder()
+                .put("standardDirectionOrder", buildOrderWithStatus(orderStatus))
+                .put("judgeAndLegalAdvisor", JudgeAndLegalAdvisor.builder().build())
+                .build())
+            .build();
+    }
```

Comment:
Little bugbear but could these be below the method below, makes more sense to me as you call `buildCaseDetailsWithHearingDetails` which calls `buildCaseDetails` which calls `buildOrderWithStatus`